OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1467 @@
+                                if (callbackList.contains(p.asObjectType) ||
+                                    (ot.startsWith("ObjectType(android") && ot.endsWith("Listener)") && ot.substring(ot.lastIndexOf("$") + 1).startsWith("On"))) {
+                                    methodCall.asVirtualMethodCall.params(i).asVar.value.asReferenceValue.upperTypeBound.filter(_.isObjectType).foreach { rt ⇒
+                                        val cf = project.classFile(rt.asObjectType)
+                                        if (cf.isDefined) {
+                                            val methods = cf.get.methods.filter(f ⇒ !f.isInitializer)
+                                            methods.foreach { method ⇒
+                                                callbackMethods += method
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+        }
+        if (callbackMethods.nonEmpty) {
+            generateCallBackEdges(m, callbackMethods)
+        }
+    }
+
+    def generateCallBackEdges(
+        m:               Method,
+        callbackMethods: Set[Method]
+    ): Unit = {
+
+        val cf = m.classFile
+        val compType = cf.thisType
+
+        if (project.classHierarchy.isSubtypeOf(compType, androidActivityOT)) {
+            val relevantLifecycleMethods = ListBuffer.empty[Method]
+            relevantLifecycleMethods ++= cf.findMethod(onStart)
+            relevantLifecycleMethods ++= cf.findMethod(onResume)
+            relevantLifecycleMethods ++= cf.findMethod(onRestart)
+            if (relevantLifecycleMethods.nonEmpty) {
+                callbackMethods.foreach { ce ⇒
+                    if (ce.classFile != m.classFile) {
+                        relevantLifecycleMethods.foreach { lm ⇒
+                            edges += ((lm, 0) -> ce)
+                        }
+                    }
+                }
+            }
+        } else if (project.classHierarchy.isSubtypeOf(compType, androidServiceOT)) {
+            val relevantLifecycleMethods = ListBuffer.empty[Method]
+            relevantLifecycleMethods ++= cf.findMethod(onStart)
+            relevantLifecycleMethods ++= cf.findMethod(onBind)
+            relevantLifecycleMethods ++= cf.findMethod(onRebind)
+            if (relevantLifecycleMethods.nonEmpty) {
+                callbackMethods.foreach { ce ⇒
+                    if (ce.classFile != m.classFile) {
+                        relevantLifecycleMethods.foreach { lm ⇒
+                            edges += ((lm, 0) -> ce)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if the intent is reachable within the processed method.
```

Comment:
Consider using different terminology. "reachable" sounds like dead vs. live code, but this doesn't seem to be what is meant here.